### PR TITLE
NON-272: Install non-associations REST client from api repository

### PR DESCRIPTION
**Pull request by ushkarev**
See [hmpps-non-associations-api#129](https://github.com/ministryofjustice/hmpps-non-associations-api/pull/129) & [hmpps-non-associations-api#132](https://github.com/ministryofjustice/hmpps-non-associations-api/pull/132).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,8 +35,7 @@
     "clean": "rm -rf assets/stylesheets build dist node_modules"
   },
   "engines": {
-    "node": "^18",
-    "npm": "^9"
```
Comment (ushkarev) [inline package.json:39]: npm v10 is the latest but we don’t hugely care; this package doesn’t do anything special

File: server/data/nonAssociationsApi.ts [hunk 2]
```diff
@@ -1,466 +1,38 @@
...
-export type Reason = typeof reasonOptions
-
-export const restrictionTypeOptions = {
-  CELL: 'Cell only',
-  LANDING: 'Cell and landing',
-  WING: 'Cell, landing and wing',
-} as const
-export type RestrictionType = typeof restrictionTypeOptions
-
 export const maxCommentLength = 240 as const
```
Comment (ushkarev) [inline server/data/nonAssociationsApi.ts:19]: limited comment lengths is not a restriction by the api, but a user interface choice, so it resides here and not in the library

File: server/data/nonAssociationsApi.ts [hunk 3]
```diff
@@ -1,466 +1,38 @@
...
-  { username: 'NON_ASSOCIATIONS_API', firstName: 'System', lastName: '' },
-  { username: 'PRISONER_MANAGER_API', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-from-nomis-migration-non-associations', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-from-nomis-migration-non-associations-1', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-to-nomis-update-non-associations', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-to-nomis-update-non-associations-1', firstName: 'System', lastName: '' },
-]
+})
 
 /**
```
Comment (ushkarev) [inline server/data/nonAssociationsApi.ts:37]: dealing with staff user lookups and non-association grouping is specific to this application and was not extracted into the shared library

**Comment by xoen** [comment]
Haven't looked at this yet but from the look of the other PR it seems like we'd have to add a GH access token to the `.npmrc`, so I'd probably add this file to this repo's `.gitignore` to reduce the risk of someone pushing their token (although, it should be a readonly token on a public repo, so in theory not catastrophic I'd say, but still)

**Comment by ushkarev** [comment]
> Haven't looked at this yet but from the look of the other PR it seems like we'd have to add a GH access token to the `.npmrc`, so I'd probably add this file to this repo's `.gitignore` to reduce the risk of someone pushing their token (although, it should be a readonly token on a public repo, so in theory not catastrophic I'd say, but still)

`.npmrc` was required in the first method i tried (using github packages) and in that case we _would_ want that file in this repo (it's in one of my commits). but then the second method (using github releases’ assets) no longer requires the `.npmrc` file so i deleted it. whether the file is here or not, we shouldn't put tokens into it, of course or bad things might happen :)

**Comment by xoen** [comment]
Nice, sorry didn't see the PR but just notice you now install the package from the public tarball, so no need to put access token in the `.npmrc` anymore 👌

**Review by xoen — DISMISSED** [review]
This LGTM, I need to have a look at the package PR but the tests passing here is a very good sign :)

**Comment by ushkarev** [comment]
(rebased without changes)

**Review by xoen — APPROVED** [review]

**Pull request merged**